sdk/search/azure-search-documents/azure/search/documents/_service/_search_service_client.py

```diff
@@ -45,6 +45,14 @@ class SearchServiceClient(HeadersMixin):  # pylint: disable=too-many-public-meth
     def __init__(self, endpoint, credential, **kwargs):
         # type: (str, AzureKeyCredential, **Any) -> None
 
+        try:
+            if endpoint.lower().startswith('http') and not endpoint.lower().startswith('https'):
```

Comment:
Is there scenario were a valid endpoint would not start by `http`?

Reply:
service only supports https